mlflow/pipelines/steps/train.py

```diff
@@ -158,6 +172,104 @@ def _run(self, output_directory):
             raw_validation_df, prediction_result, self.target_col
         )
 
+        from mlflow.entities import ViewType
+
+        mlflow_client = MlflowClient()
+        exp_id = _get_experiment_id()
+
+        primary_metric_greater_is_better = self.metric_greater_is_better[self.primary_metric]
+
+        primary_metric_order = "DESC" if primary_metric_greater_is_better else "ASC"
+
+        search_max_results = 100
+        search_result = mlflow_client.search_runs(
+            experiment_ids=exp_id,
+            run_view_type=ViewType.ALL,
+            max_results=search_max_results,
+            order_by=[f"metrics.{self.primary_metric} {primary_metric_order}"],
+        )
+
+        primary_metric_name = _get_primary_metric(self.step_config)
+        custom_metric_names = [cm["name"] for cm in _get_custom_metrics(self.step_config)]
+        metric_names = self.metric_greater_is_better.keys()
+        metric_keys = [f"{metric_name}_on_data_validation" for metric_name in metric_names]
```

Comment:
I'm not sure `metric_keys` simplifies the code much.

Reply:
`metric_keys` is used in multiple places below, which is why it's helpful to define it up here